### ISSUE TITLE
Svelte: connection forwarding in handle / on:connect issues

### ISSUE THREAD

**Issue by 1bye** [xyflow/xyflow#3662]
Hi! Please add feature connection forwarding in handle, thanks
```js
<Handle id="some-id" class="..." let:connected>
    {#if connected}
        // show some content
    {/if}
</Handle>
``` 
Thanks

_Originally posted by @1bye in https://github.com/xyflow/xyflow/issues/3618#issuecomment-1815390238_

**Comment by moklick** [comment]
We can see that users are struggling with `on:connect` for `<Handle />` components. Currently it's only useful for forwarding the event so that you can use `on:connect` with the `<SvelteFlow />` component. 

Users want to know when a handle gets connected or disconnected. Currently I can see three possible APIs for this:

1. as explained above, we pass information back to the parent with something like `let:connected` or `let:connectionState`
2. we add `onconnect` and `ondisconnect` handlers that get called for all handles (currently this only happens for the the handle where a connection starts)
3. we add a helper hook to provide that information:
```js 
const handleConnectionState = useHandleConnectionState(handleRef);
```

I am not sure what would be the best way to go here, but opt 2. seems straight forward. wdyt?

**Comment by futurGH** [comment]
Personally, either of options 1 and 2 would be great to have, though as a new user, 2 seems more in line with how I'd already expect things to work.

Currently using edges array reactivity tracking and it feels super wasteful for something like this.

**Comment by moklick** [comment]
Thanks for the feedback. That said, option 3 wouldn't look like a good API for you?

**Comment by adamdva** [comment]
I think option #2 is more consistent with the handlers on the SvelteFlow element.
Option #3 works, but does require an additional bind to get the ref (which we're already doing to apply dynamic style classes.

We could work with any of the options, but #2 would get my top vote.

**Comment by sincethestudy** [comment]
I would expect to be able to do on:connect for both source and target, which would be option 2. For my application, I care about target knowing when connection happens, which I can't do right now.

**Comment by PapaTemporal** [comment]
I have a PR for this https://github.com/xyflow/xyflow/pull/3636
It basically just needs on:connect on the source and destination handle, but here is an example of it's use:
https://github.com/PapaTemporal/PyFlowAutomator/blob/main/src/lib/nodes/exec/HttpRequest.svelte

If on:connect isn't set, then nothing happens and if they just want a trigger, the on:connect still needs to be added with nothing in it.

**Comment by futurGH** [comment]
I'd say no, unless there's a way to distinguish between a user-initiated connection and a preexisting one in the event.

**Comment by moklick** [comment]
fixed in 0.0.29 https://svelteflow.dev/whats-new/2023-12-13

You can now pass `onconnect`, `onconnectstart` and `onconnectend` to the `<SvelteFlow />` component and there is no need to forward anything. In addition the `<Handle />` component has it's own `onconnect` and `ondisconnect` handlers that work for all handles. Regardless if its a source or target.
